libraries/adaptive-expressions/src/converters/enumExpressionConverter.ts

```diff
@@ -15,31 +15,45 @@ type Input<T> = T | string | Expression;
  * `string` to json [EnumExpression](xref:adaptive-expressions.EnumExpression) converter.
  */
 export class EnumExpressionConverter<T> {
-    private _enumValue: unknown;
+    private readonly lowercaseIndex: Record<string, string>;
 
     /**
      * Initializes a new instance of the [EnumExpressionConverter](xref:adaptive-expressions.EnumExpressionConverter) class.
+     *
      * @param enumValue The enum value of the `string` to convert.
      */
-    public constructor(enumValue: unknown) {
-        this._enumValue = enumValue;
+    public constructor(private readonly enumValue: unknown) {
+        this.lowercaseIndex = Object.keys(enumValue || {}).reduce((acc, key) => {
+            acc[key.toLowerCase()] = key;
+
+            return acc;
+        }, {});
     }
 
     /**
      * Converts a `string` into an [EnumExpression](xref:adaptive-expressions.EnumExpression).
+     *
      * @param value `string` to convert.
      * @returns The [EnumExpression](xref:adaptive-expressions.EnumExpression).
      */
     public convert(value: Input<T> | EnumExpression<T>): EnumExpression<T> {
         if (value instanceof EnumExpression) {
             return value;
         }
-        if (typeof value == 'string') {
-            if (Object.prototype.hasOwnProperty.call(this._enumValue, value)) {
-                return new EnumExpression<T>(this._enumValue[value as string]);
+
+        if (typeof value === 'string') {
+            let enumValue = this.enumValue[value];
+            if (enumValue === undefined) {
+                enumValue = this.enumValue[this.lowercaseIndex[value]];
+            }
+
+            if (enumValue !== undefined) {
+                return new EnumExpression<T>(enumValue as Input<T>);
```

Comment:
```suggestion
        const enumValue = this.enumValue[value] ?? this.enumValue[this.lowercaseIndex[value]];

        if (enumValue !== undefined) {
            return new EnumExpression<T>(enumValue as Input<T>);
```

Reply:
Sadly adaptive expressions is pinned on and older Typescript version. That was my first try!

Reply:
That’s a bummer… A dependent’s typescript dependency, right?

Reply:
Yep, I don't think our version of antlr works with newer Typescript versions.